docs/weave-tutorial/03-codec.md

```diff
@@ -0,0 +1,172 @@
+---
+id: codec
+title: Codec
+sidebar_label: Codec
+---
+
+> [PR#1](https://github.com/iov-one/tutorial/pull/1): _Create order book models_
+
+Weave defines simple `Marshaller` and `Persistent` interface standards. These interfaces are automatically implemented by any code autogenerated by protoc from protobuf files, and we recommend using .proto files to specify the serialization format for any persistent data in our application (internal state as well as transactions and messages). However, if you have never worked with protobuf, this might be a bit of a challenge, so we explain a simple workflow that we use in weave based projects.
+
+## Create Codec file
+
+The first thing is: imagine the shape of your classes. This `Codec` should be defined in [proto3](https://developers.google.com/protocol-buffers/docs/proto3) syntax. There are a number of different `int encodings`, `byte slices`, `strings`, and nested structures. And fields may be repeated. So forget complex types with methods now and just focus on the actual data structure.
+
+Codec is the first the component that needs to be designed. Keep in mind that this part is the most important since whole module will depend on _codec_. You can think codec it as _model_ in mvc pattern. Yet it is not simple as model. Codec defines the whole application state models and more. On this file we would not go far as explaining every proto message. One example for each message and state would be enough. Further more if you want to see all the implementation, you can check out [tutorial](https://github.com/iov-one/tutorial/blob/master/x/orderbook/codec.proto)
```

Comment:
I did my best :sweat: Believe me I am improving fast. You should have seen my first technical writings  :smile: Anyways Jim(Editor) will take look at all the docs.

Reply:
No worries. I see those (I think) mistakes, because that is what I do a lot. Those constructs do not exist in polish, so I have hard time remembering about them. Only recently I started to pay attention to grammar a bit more :sweat_smile: